### ISSUE TITLE
Element doesn't respect system font scaling

### ISSUE THREAD

**Issue by thefirstofthe300** [vector-im/element-android#5687]
### Steps to reproduce

1) Set system font size to any value (smallest for greatest contrast)
2) Open Element and observe current font size
3) Set system font size to another value (largest for greatest contrast
4) Open Element and observe that the font sizes have stayed the same (with the notable exception of the chips used to denote @'ing someone

### Outcome

Font sizes don't change 

### Will you send logs?

No

**Comment by fedrunov** [comment]
We have currently app-wide setting to define font scale (Settings->Preferences->Font size) and the problem is that on app start we always set font scale for the app from that setting. So if system setting is the "largest" text scale but in-app setting is "normal", the resulting applying scale will be "normal", and if user change system setting we ignore it, applying only whatever set in the app settings. The only moment when we sync with system property is first start of the app.-  we get system property and apply it to the app setting. 

I think we should have "system default" setting as default one, until user manually change it to something specific in the app settings. If "system default" is set, app will use system settings, reflecting any changes in them. @daniellekirkwood what do you think?

This is how in-app font scale setting dialog look like now:
![Screenshot 2022-05-24 at 12 18 43](https://user-images.githubusercontent.com/66663241/170010671-5f09fb5e-d376-47e4-a880-dd4c637800b3.png)

**Comment by daniellekirkwood** [comment]
Thanks, @fedrunov !!

Let's go ahead and add "Use system default" to the top of this list with a radio button. We'll use font size "Normal" for the label.

**Comment by fedrunov** [comment]
Here how it looks like:

![image](https://user-images.githubusercontent.com/66663241/170479509-47161c65-260f-4a9b-b4f6-0482f594f184.png)

I think it would be better if list will contain only scale factor items and "use system default" will be on top as a checkbox. When it's checked list will be "readonly" and item, corresponding to system setting will be selected. When it's unchecked list will be interactive and user could select an option to override system setting for element.

**Comment by fedrunov** [comment]
@niquewoodhouse, @amshakal what do you think?

**Comment by amshakal** [comment]
Hey! I agree with your solution, the only thing I'll add is a line seperating system default from the rest of the options. Happy to create a visual representation of it if needed.
